File: JetpacReloaded/C# version/JetpacReloaded/Rocket.cs
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;

namespace JetpacReloaded
{
    public class Rocket : BaseObject
    {
        public Rocket(int x, int y, Texture2D image, int frame, int width, int height)
        {
            m_image = image;
            m_frame = frame;
            m_width = width;
            m_height = height;
            m_rect = new Rectangle(m_frame * m_width, 0, m_width, m_height);
            m_screenLocation = new Vector2(x, y);
        }

        public void Update(int x, int y)
        {
            m_screenLocation.X = x;
            m_screenLocation.Y = y;
        }

        public bool LowerSectionOne()
        {
            m_screenLocation.X = 422;
            if (m_screenLocation.Y < 383)
            {
                m_screenLocation.Y++;
            }
            return m_screenLocation.Y >= 383;
        }

        public bool LowerSectionTwo()
        {
            m_screenLocation.X = 422;
            if (m_screenLocation.Y < 323)
            {
                m_screenLocation.Y++;
            }
            return m_screenLocation.Y >= 323;
        }

        public bool TakeOff()
        {
            m_screenLocation.Y -= 1.5f;
            return m_screenLocation.Y > -200;
        }

        new public void Draw(SpriteBatch spriteBatch)
        {
            spriteBatch.Draw(m_image, m_screenLocation, m_rect, Color.White, 0, Vector2.Zero, 1.0f, 0, 0);
        }

        public Rectangle RocketRect
        {
            get { return new Rectangle((int)m_screenLocation.X, (int)m_screenLocation.Y, m_width, m_height); }
        }
    }
}

File: JetpacReloaded/Typescript version/WebGLCanvas/ts/rocket.ts
import { Color } from "../ts_lib/color.js";
import { Rectangle } from "../ts_lib/rectangle.js";
import { SpriteBatch } from "../ts_lib/spritebatch.js";
import { Texture2D } from "../ts_lib/texture2D.js";
import { Vector2 } from "../ts_lib/vector2.js";
import { BaseObject } from "./baseobject.js";

export class Rocket extends BaseObject {
    constructor(x: number, y: number, image: Texture2D, frame: number, width: number, height: number) {
        super();
        this.m_image = image;
        this.m_frame = frame;
        this.m_width = width;
        this.m_height = height;
        this.m_rect = new Rectangle(this.m_frame * this.m_width, 0, this.m_width, this.m_height);
        this.m_screenLocation = new Vector2(x, y);
    }
    public Update(x: number, y: number): void {
        this.m_screenLocation.X = x;
        this.m_screenLocation.Y = y;
    }
    public LowerSectionOne(): boolean {
        this.m_screenLocation.X = 422;
        if (this.m_screenLocation.Y < 383) {
            this.m_screenLocation.Y++;
        }
        return this.m_screenLocation.Y >= 383;
    }
    public LowerSectionTwo(): boolean {
        this.m_screenLocation.X = 422;
        if (this.m_screenLocation.Y < 323) {
            this.m_screenLocation.Y++;
        }
        return this.m_screenLocation.Y >= 323;
    }
    public TakeOff(): boolean {
        this.m_screenLocation.Y -= 1.5;
        return this.m_screenLocation.Y > -200;
    }
    public Draw(spriteBatch: SpriteBatch): void {
        spriteBatch.Draw(this.m_image, this.m_screenLocation, this.m_rect, Color.White, 0, Vector2.Zero, 1.0, 0, 0);
    }
    public get RocketRect(): Rectangle {
        return new Rectangle(this.m_screenLocation.X, this.m_screenLocation.Y, this.m_width, this.m_height);
    }
}